examples/integration-tutorials/routing-requests-based-on-message-content/guide/content_based_routing.bal

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/log;
+
+// Service to reserve appointments
+@http:ServiceConfig {
+    basePath: "/healthcare"
+}
+service contentBasedRouting on new http:Listener(9080) {
+
+    // Reserve appointments on the type of "category"
+    @http:ResourceConfig {
+        methods: ["POST"],
+        path: "/categories/{category}/reserve"
+    }
+    resource function CBRResource(http:Caller outboundEP, http:Request req, string category) {
+
+        // Request message payload
+        var jsonMsg = req.getJsonPayload();
+        if (jsonMsg is json) {
+            string hospitalDesc = jsonMsg["hospital"].toString();
+            string doctorName = jsonMsg["doctor"].toString();
+            string hospitalName = "";
+
+            // Endpoint URL of the backend service
+            http:Client locationEP = new("http://localhost:9090");
+            http:Response|error clientResponse;
+            if (hospitalDesc != "") {
+                match hospitalDesc {
+                    "grand oak community hospital" => hospitalName = "grandoaks";
+                    "clemency medical center" => hospitalName = "clemency";
+                    "pine valley community hospital" => hospitalName = "pinevalley";
```

Comment:
I think it's better to handle the default case as well where we respond with a bad request to the client. WDYT?

Reply:
Added